File: app/src/main/java/it/gruppoinfor/home2work/utils/LatLngConverter.java
package it.gruppoinfor.home2work.utils;

import android.arch.persistence.room.TypeConverter;

import com.google.android.gms.maps.model.LatLng;

import java.util.Locale;

public class LatLngConverter {

    @TypeConverter
    public static LatLng fromString(String stringLatLng) {
        String[] strings = stringLatLng.split(",");
        double lat = Double.parseDouble(strings[0]);
        double lng = Double.parseDouble(strings[1]);
        return new LatLng(lat, lng);
    }

    @TypeConverter
    public static String toString(LatLng latLng) {
        return String.format(Locale.ROOT, "%1$.8f,%2$.8f", latLng.latitude, latLng.longitude);
    }
}

File: App/src/main/java/it/gruppoinfor/home2work/utils/LatLngConverter.kt
package it.gruppoinfor.home2work.utils


import android.arch.persistence.room.TypeConverter
import it.gruppoinfor.home2workapi.model.LatLng

class LatLngConverter {

    @TypeConverter
    fun fromString(stringLatLng: String): LatLng {

        val strings = stringLatLng.split(",")
        val lat = strings[0].toDouble()
        val lng = strings[1].toDouble()

        return LatLng(lat, lng)
    }

    @TypeConverter
    fun toString(latLng: LatLng): String {

        return "%.8f,%.8f".format(latLng.lat, latLng.lng)
    }
}
